domain/access/state/permission.go

```diff
@@ -273,46 +369,59 @@ func (st *PermissionState) ReadAllUserAccessForUser(ctx context.Context, subject
 		return nil, errors.Trace(err)
 	}
 	var (
-		permissions []dbPermission
 		users       []dbPermissionUser
+		permissions []dbPermission
 	)
 	query := `
-SELECT (u.uuid, u.name, u.display_name, u.created_at, u.disabled) AS (&dbPermissionUser.*),
+SELECT (u.uuid, u.name, u.display_name, u.external, u.created_at, u.disabled) AS (&dbPermissionUser.*),
        creator.name AS &dbPermissionUser.created_by_name,
-       (p.uuid, p.grant_on, p.grant_to, p.access_type, p.object_type) AS (&dbPermission.*)
+       (p.*) AS (&dbPermission.*)
 FROM   v_user_auth u
        JOIN user AS creator ON u.created_by_uuid = creator.uuid
-       JOIN v_permission p ON u.uuid = p.grant_to
+       LEFT JOIN v_permission p ON u.uuid = p.grant_to
 WHERE  u.removed = false
        AND u.name = $userName.name
 `
 
 	uName := userName{Name: subject}
 
-	queryStmt, err := st.Prepare(query, dbPermissionUser{}, dbPermission{}, uName)
+	queryStmt, err := st.Prepare(query, uName, dbPermissionUser{}, dbPermission{})
 	if err != nil {
 		return nil, errors.Annotate(err, "preparing select all access for user query")
 	}
 
+	var externalPerms []dbPermission
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err = tx.Query(ctx, queryStmt, uName).GetAll(&permissions, &users)
+		err = tx.Query(ctx, queryStmt, uName).GetAll(&users, &permissions)
 		if errors.Is(err, sql.ErrNoRows) {
 			return errors.Annotatef(accesserrors.UserNotFound, "%q", subject)
 		} else if err != nil {
 			return errors.Annotatef(err, "getting user with name %q", subject)
 		}
+
+		if users[0].External {
```

Comment:
I know that we check for the error message `sql.ErrNoRows`, but to prevent a panic can we also ensure that we validate the length is greater than 0.

Reply:
Added an `|| len(users) == 0` to the error check from the `GetAll` above

Reply:
By the way, SQLair does guarantee that a `ErrNoRows` will be returned in this case.